Source/Extensions/CAShapeLayerExtension.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  CAShapeLayerExtension.swift
+//  
+//
+//  Created by Ehab Alsharkawy on 14/11/2022.
+//
+
+import UIKit
+
+extension CAShapeLayer {
+
+    /// Add Edges and corners with style
+    /// - Parameter style: Protocol that provide which edge or corner will be added with style
+    func createCustomBorder(with style: ElementBorderStyle) {
+
+        // if path not nil then it's already added so no need to draw lines or curves again
+        guard self.path == nil else { return }
+
+        // always fill is clear color
+        fillColor = UIColor.clear.cgColor
+
+        lineWidth = style.borderWidth
+        let rect = bounds.insetBy(dx: style.cornerRadius / 2,
+                                  dy: style.cornerRadius / 2)
+        let path = UIBezierPath()
+        applyBorders(with: path, rect: rect, style: style)
+        updateCorners(with: path, rect: rect, style: style)
```

Comment:
I think these 2 methods should read the same

```suggestion
        applyBorders(with: path, rect: rect, style: style)
        applyCorners(with: path, rect: rect, style: style)
```